pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java

```diff
@@ -277,6 +277,7 @@ private void phaseTwoLoop(RawReader reader, MessageId to, Map<String, MessageId>
                                     promise.complete(null);
                                 }
                             });
+                            return;
```

Comment:
What if some of the previous addToCompactedLedger aync write fail later, but the last id's addToCompactedLedger is complete first? Do we mark the `promise` as successful here? 

Don't we need to wait for all outstanding addToCompactedLedger to be complete before making `promise` complete?

Reply:
You are right, but this PR is to fix phase two can't be stopped. We can open another PR to fix this issue.

Reply:
@heesung-sn I opened https://github.com/apache/pulsar/pull/21067 to fix it.